### ISSUE TITLE
API Types Library

### ISSUE THREAD

**Issue by Eralyne** [DasWolke/SnowTransfer#39]
I wanted to ask if there was any reason this library is using a custom typings library from https://github.com/AmandaDiscord/DiscordTypings rather than something more widely used and updated such as https://github.com/discordjs/discord-api-types.

I'm having trouble implementing certain functionalities because I'm getting type errors between the two and I can't cast some of the objects (easily....) because it seems the typings this library uses are outdated.

An example is `ApplicationCommandBase` from DiscordTypings appears to be older. According to the [docs](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-structure), there should be a required id, version, and among other things that the `ApplicationCommandBase` does not seem to include but `APIApplicationCommand` from the more [widely-used library](https://github.com/discordjs/discord-api-types) has the type that is identical to the discord API docs.

I like the way this library acts as a lightweight rest client, I'm just wondering if this is something intended, if I am missing something, or I should use something else/fork and adjust.

**Comment by PapiOphidian** [comment]
I'm the owner and maintainer of the @AmandaDiscord organization. At the time of implementation and rewriting this library, discord.js was not up to date by a considerable margin with what the docs had for whatever reason. Not like it really matters to me. I decided to make my own so that I could update at my earliest convenience as I was very much wanting to keep the ball rolling so to speak as I was rewriting this lib and others to TypeScript.

I ended up almost completely burning myself out on Discord related dev work trying to keep up to date with the Discord docs github updating discord-typings, a cacheless TypeScript "remake" of discord.js I called ThunderStorm, as well as my two public bots, SnowTransfer, CloudStorm and RainCache all by myself.

I was working on moving over to discord-api-types and deprecating discord-typings as I no longer wish to try to keep up with so many commits, discussions, etc on Discord's end, so expect that in the coming few days once I actually get a day off work 🫠
